src/components/cleaner.ts

```diff
@@ -112,27 +151,9 @@ export class Cleaner {
                 }
             }
         }
-    }
 
-    private async cleanGlobFolders(globs: string[], pathPairsToRemove: [string, string][]) {
-        // All glob patterns EXPLICITLY given for folders, the glob should be end with a slash (`path.sep`) and the last component should not contain globstar `**`
-        // Positive examples: ['abc/', 'abc*/', '**/abc*/', 'abc/**/def/', 'abc/**/def*/']
-        // Negative examples:
-        //   - not end with a slash: ['abc', 'abc*', 'abc/**/def*']
-        //   - contain globstar `**` in the last component: ['**', '**/', 'abc/**', 'abc/**/', 'abc/def**/', 'abc/d**ef/']
-        const explicitFolderGlobs: string[] = globs.filter(globType => globType.endsWith(path.sep) && !path.basename(globType).includes('**'))
-        const matchExplicitFolderGlobs = (folder: string): boolean => explicitFolderGlobs.some(globType => minimatch(folder, globType))
-        // Each folder path should match at least one EXPLICIT folder glob pattern
-        // NOTE: All elements in this array is a directory because they are globed by patterns with tailing slashes.
-        const explicitFolderPathsToRemove: string[] = pathPairsToRemove.filter(
-            ([globPathResult, realPath]) => realPath.endsWith(path.sep) && matchExplicitFolderGlobs(globPathResult)
-        ).map(([_, realPath]) => realPath)
-        // Sort by descending dictionary order, make sure the children folders are sorted before the parents
-        // For example: [..., 'out/folder1/folder2/', 'out/folder1/', ...] ('out/folder1/folder2/' > 'out/folder1/' in directory order)
-        .sort((a, b) => b.localeCompare(a))
-
-        // Remove empty folders EXPLICITLY specified
-        for (const folderRealPath of explicitFolderPathsToRemove) {
+        // Then remove empty folders EXPLICITLY specified
+        for (const folderRealPath of explicitFolders) {
             try {
                 // We are sure that this is a folder because it's globed by pattern with a trailing slash
```

Comment:
I guess the name `explicitFolders` is sufficiently meaningful to know that it only contains folders.